src/lib/y2storage/volume_specification.rb

```diff
@@ -0,0 +1,202 @@
+    #   from the control file) that will be used to plan the Btrfs subvolumes
+    attr_accessor :subvolumes
+
+    # @return [String] default btrfs subvolume path
+    attr_accessor :btrfs_default_subvolume
+
+    # @return [Numeric] order to disable volumes if needed to make the initial proposal
+    attr_accessor :disable_order
+
+    alias_method :proposed?, :proposed
+    alias_method :proposed_configurable?, :proposed_configurable
+    alias_method :adjust_by_ram?, :adjust_by_ram
+    alias_method :adjust_by_ram_configurable?, :adjust_by_ram_configurable
+    alias_method :snapshots?, :snapshots
+    alias_method :snapshots_configurable?, :snapshots_configurable
+
+    def initialize(volume_features)
+      @volume_features = volume_features
+      apply_defaults
+      load_features
+    end
+
+  private
+
+    attr_reader :volume_features
+
+    def apply_defaults
+      @fs_types   ||= []
+      @subvolumes ||= []
+      @max_size   ||= DiskSize.unlimited
+    end
+
+    # For some features (i.e., fs_types and subvolumes) fallback values could be applied
+    def load_features
+      {
+        mount_point:                :string,
+        proposed:                   :boolean,
+        proposed_configurable:      :boolean,
+        fs_types:                   :list,
+        fs_type:                    :string,
+        adjust_by_ram:              :boolean,
+        adjust_by_ram_configurable: :boolean,
+        fallback_for_min_size:      :string,
+        fallback_for_max_size:      :string,
+        fallback_for_max_size_lvm:  :string,
+        fallback_for_weight:        :string,
+        snapshots:                  :boolean,
+        snapshots_configurable:     :boolean,
+        btrfs_default_subvolume:    :string,
+        desired_size:               :size,
+        min_size:                   :size,
+        max_size:                   :size,
+        max_size_lvm:               :size,
+        snapshots_size:             :size,
+        weight:                     :integer,
+        disable_order:              :integer,
+        subvolumes:                 :subvolumes
+      }.each do |feature, type|
+        type = nil if [:string, :boolean, :list].include?(type)
+        loader = type.nil? ? "load_feature" : "load_#{type}_feature"
+        send(loader, feature, source: volume_features)
+      end
+
+      apply_fallbacks
+    end
+
+    def fs_types=(types)
+      @fs_types = types.map { |t| validated_fs_type(t) }
+    end
+
+    def fs_type=(type)
+      @fs_type = validated_fs_type(type)
+    end
+
+    def validated_fs_type(type)
+      raise(ArgumentError, "Filesystem cannot be nil") unless type
+      Filesystems::Type.find(type.downcase.to_sym)
+    end
+
+    def root?
+      mount_point && mount_point == "/"
+    end
+
+    def apply_fallbacks
+      apply_subvolumes_fallback
+      apply_fs_types_fallback
+    end
+
+    def apply_subvolumes_fallback
```

Comment:
Mmm, this makes impossible to specify that you want an empty list of subvolumes for root. Is that what we want?